pkg/capi/edit/ClusterConfig.vue

```diff
@@ -0,0 +1,429 @@
+    },
+    async saveOverride() {
+      if ( this.errors ) {
+        clear(this.errors);
+      }
+      await this.value.save();
+
+      return this.done();
+    },
+
+    initSpecs() {
+      if ( !this.value.spec ) {
+        set(this.value, 'spec', { });
+      }
+      if ( !this.value.spec.topology ) {
+        set(this.value.spec, 'topology', clone(defaultTopologyConfig));
+      }
+
+      if ( !this.value.spec.clusterNetwork ) {
+        set(this.value.spec, 'clusterNetwork', clone(defaultClusterNetwork));
+      }
+
+      if ( !this.value.spec.controlPlaneEndpoint ) {
+        set(this.value.spec, 'controlPlaneEndpoint', clone(defaultCPEndpointConfig));
+      }
+    },
+
+    stepClusterClassReady() {
+      const step = this.addSteps.find(s => s.name === 'stepClusterClass');
+
+      this.$set(step, 'ready', this.stepClusterClassRequires);
+    },
+
+    stepConfigurationReady() {
+      const step = this.addSteps.find(s => s.name === 'stepConfiguration');
+
+      this.$set(step, 'ready', this.stepConfigurationRequires);
+    },
+    stepVariablesReady() {
+      const step = this.addSteps.find(s => s.name === 'stepVariables');
+
+      this.$set(step, 'ready', this.stepVariablesRequires);
+    },
+
+    cancelCredential() {
+      if ( this.$refs.cruresource ) {
+        this.$refs.cruresource.emitOrRoute();
+      }
+    },
+    validateVariables(neu: Boolean) {
+      this.variablesReady = neu;
+      this.stepVariablesReady();
+    },
+    cancel() {
+      this.$router.push({
+        name:   'c-cluster-manager-capi',
+        params: {},
+      });
+    },
+    done() {
+      this.$router.push({
+        name:   'c-cluster-manager-capi',
+        params: {},
+      });
+    },
+    machineDeploymentsChanged(val: Worker) {
+      this.set(this.value.spec.topology.workers, 'machineDeployments', val);
+      this.stepConfigurationReady();
+    },
+    machinePoolsChanged(val: Worker) {
+      this.set(this.value.spec.topology.workers, 'machinePools', val);
+      this.stepConfigurationReady();
+    },
+    cpEndpointHostChanged(val: String) {
+      this.set(this.value.spec.controlPlaneEndpoint, 'host', val);
+      this.stepConfigurationReady();
+    },
+    cpEndpointPortChanged(val: Number) {
+      this.set(this.value.spec.controlPlaneEndpoint, 'port', val);
+      this.stepConfigurationReady();
+    },
+    clickedType(obj: Object) {
+      this.clusterClassObj = this.clusterClasses.find(x => x.metadata.name === obj.id);
+      this.set(this.value.spec.topology, 'class', obj.id);
+      this.stepClusterClassReady();
+    }
+  }
+});
+</script>
+<template>
+  <Loading v-if="$fetchState.pending" />
+  <CruResource
+    v-else
+    ref="cruresource"
+    :mode="mode"
+    :validation-passed="fvFormIsValid"
+    :show-as-form="true"
+    :resource="value"
+    :errors="errors"
+    :cancel-event="true"
+    :done-route="doneRoute"
+    :apply-hooks="applyHooks"
+    :generate-yaml="generateYaml"
```

Comment:
This is breaking the 'edit as yaml' button. At a glance removing this seems to work, but did you hit an issue with the default yaml-generating function in CruResource? We should hide the 'edit as yaml' button for now if that's the case.

Reply:
removed